### ISSUE TITLE
enhancement: update the default rate limiting rejection code from 503 to 429

### ISSUE THREAD

**Issue by kayx23** [apache/apisix#9255]
### Current Behavior

Previous discussion can be found here: https://www.mail-archive.com/dev@apisix.apache.org/msg05298.html

Notably the following: 
> The rate-limit plugin currently defaults to 503 HTTP response code for requests exceeding the configured rate limit. HTTP `429 Too Many Requests` seems to be a better fit: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429 
> 
> 503 was initially used because Nginx's limit_req_status also uses 503 as the default. However, Nginx made its decision long before 429 became popular.

This isn't a bug per se, but an enhancement, which has been noted should be announced as a breaking change. 

Plugins in question:
* limit-conn
* limit-count
* limit-req

### Steps to Reproduce

https://docs.api7.ai/apisix/getting-started/rate-limiting

### Environment

APISIX version 3.2

**Comment by kayx23** [comment]
cc @nfrankel @moonming

**Comment by nfrankel** [comment]
I stand by my initial point of view: we should strive to uphold the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), which means return the expected code: 429

**Comment by smileby** [comment]
> I stand by my initial point of view: we should strive to uphold the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), which means return the expected code: 429
> 
> However, because it's a breaking change, we should defer the implementation until Apache APISIX 4.0

I don’t think so. The ‘rejected_code’ response code itself supports user customization.

**Comment by nfrankel** [comment]
I think you misunderstand my point.

I'm not saying that it's not possible, I'm saying that the default is wrong.

**Comment by smileby** [comment]
I ithink you are right
